src/parkapi_sources/converters/apcoa/mapper.py

```diff
@@ -0,0 +1,83 @@
+"""
+Copyright 2024 binary butterfly GmbH
+Use of this source code is governed by an MIT-style license that can be found in the LICENSE.txt.
+"""
+
+from datetime import datetime, timezone
+
+from parkapi_sources.models import StaticParkingSiteInput
+from parkapi_sources.models.enums import PurposeType
+
+from .validators import ApcoaNavigationLocationType, ApcoaOpeningHoursWeekdays, ApcoaParkingSiteInput, ApcoaParkingSpaceType
+
+
+class ApcoaMapper:
+    @staticmethod
+    def map_static_parking_site(apcoa_input: ApcoaParkingSiteInput) -> StaticParkingSiteInput:
+        static_parking_site_input = StaticParkingSiteInput(
+            uid=str(apcoa_input.CarParkId),
+            name=apcoa_input.CarparkLongName if apcoa_input.CarparkLongName else apcoa_input.CarparkShortName,
+            lat=next(
+                iter(
+                    location_type_input.GeoCoordinates.Latitude
+                    for location_type_input in apcoa_input.NavigationLocations
+                    if location_type_input.LocationType == ApcoaNavigationLocationType.DEFAULT
+                )
+            ),
+            lon=next(
+                iter(
+                    location_type_input.GeoCoordinates.Longitude
+                    for location_type_input in apcoa_input.NavigationLocations
+                    if location_type_input.LocationType == ApcoaNavigationLocationType.DEFAULT
+                )
+            ),
+            purpose=PurposeType.CAR,
+            type=apcoa_input.CarparkType.Name.to_parking_site_type_input(),
+            has_realtime_data=False,  # TODO: change this as soon as Apcoa API offers realtime data
+            static_data_updated_at=datetime.now(tz=timezone.utc),
+        )
+        if apcoa_input.CarParkWebsiteURL:
+            static_parking_site_input.public_url = apcoa_input.CarParkWebsiteURL
+
+        if apcoa_input.Address.Street and apcoa_input.Address.Zip and apcoa_input.Address.City:
+            static_parking_site_input.address = f'{apcoa_input.Address.Street}, {apcoa_input.Address.Zip} {apcoa_input.Address.City}'
+
+        num_of_weekdays = len(
+            [
+                opening_hours_input.Weekday
+                for opening_hours_input in apcoa_input.OpeningHours
+                if opening_hours_input.Weekday in ApcoaOpeningHoursWeekdays.Weekdays
+                and opening_hours_input.OpeningTimes == ApcoaOpeningHoursWeekdays.OpeningTime
+            ]
+        )
+        if num_of_weekdays == 7:
```

Comment:
As the data is well structured, you could do a better mapping to OSM format to get the other opening times, too.

Reply:
Could you please elaborate on what the values of other OSM-Formats should be? i have added some but not sure if they are right.

Reply:
Voila: https://wiki.openstreetmap.org/wiki/Key:opening_hours